### DISCUSSION TITLE
What is does it mean "compile node_modules?"

### DISCUSSION THREAD

**Discussion by nyngwang** [babel/babel#14408] [Q&A] [answered]
As title, which is a part of the questions from the [docs](https://babeljs.io/docs/en/configuration#whats-your-use-case):

> You want to compile node_modules?

Could someone provide an example? Does it mean that Babel would also transform code imported from `node_modules`?

I need to know this in order to understand when to use `.babelrc` or `babel.config.json/js`.

**Comment by nicolo-ribaudo** [comment, marked as answer]
Yes, to be more specific that sentence might be "You want to compile files from the node_modules folder?"

**Reply by nyngwang** [reply]
If I use `.babelrc`, then it won't transform the code imported from `node_modules`? But `babe.config.js` will do?

**Reply by nicolo-ribaudo** [reply]
There are two levels here:
1. which files are handled by Babel
2. which config applies to each file

(1) is not decided by Babel, but by whoever is calling Babel. For example, if you are using `babel-loader` you can configure it using Webpack's `include` and `exclude` options for loaders.
(2) programmatic options and `babel.config.js` option affect every file "seen" by Babel; `.babelrc.js` options only affect files in the same package (i.e. `package.json` files create a "limit" that `.babelrc.js` cannot cross).

In general people should configure (1) to only pass to Babel the files that they want to transform, and use `babel.config.js`. If they have an advanced setup where they need a package-specific plugin/preset/option, they can add a `.babelrc.js` file.
